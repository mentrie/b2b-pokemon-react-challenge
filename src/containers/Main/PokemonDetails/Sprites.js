import React from 'react';

const Sprites = ({ sprites }) => {
  const spritesList = Object.keys(sprites).filter(key => sprites[key]);

  if (spritesList.length > 0) {
    const spritesToRender = spritesList.map((sprite, index) => (
      <div key={index} className="sprite__item">
        <div className="sprite__background__cover" style={{backgroundImage: `url(${sprites[sprite]})`}} />
        <img className="sprite__picture" alt={`sprint-${index}`} src={sprites[sprite]} />
      </div>
    ));

    return (
      <div className="sprites__wrapper">
        <h3 className="sprites__title">Sprites</h3>
        <div className="sprites__list">
          {spritesToRender}
        </div>
      </div>
    );
  }
};

export default Sprites;
