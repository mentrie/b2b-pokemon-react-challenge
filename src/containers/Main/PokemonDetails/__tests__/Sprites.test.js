import React from 'react';
import Sprites from '../Sprites';

test('renders nothing', () => {
  const props = {
    sprites: {}
  };
  const component = shallow(<Sprites {...props} />);

  expect(component).toMatchSnapshot();
  expect(component.find('.sprites__wrapper').length).toBe(0);
});

test('renders sprites component', () => {
  const props = {
    sprites: {
      1: 'https://fakeurl.com/image.png'
    }
  };
  const component = shallow(<Sprites {...props} />);
  expect(component).toMatchSnapshot();
  expect(component.find('.sprites__wrapper').length).toBe(1);
});
