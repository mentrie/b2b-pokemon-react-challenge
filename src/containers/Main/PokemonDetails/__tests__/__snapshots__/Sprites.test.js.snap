// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders nothing 1`] = `""`;

exports[`renders sprites component 1`] = `
<div
  className="sprites__wrapper"
>
  <h3
    className="sprites__title"
  >
    Sprites
  </h3>
  <div
    className="sprites__list"
  >
    <div
      className="sprite__item"
      key="0"
    >
      <div
        className="sprite__background__cover"
        style={
          Object {
            "backgroundImage": "url(https://fakeurl.com/image.png)",
          }
        }
      />
      <img
        alt="sprint-0"
        className="sprite__picture"
        src="https://fakeurl.com/image.png"
      />
    </div>
  </div>
</div>
`;
